frontend/lib/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -149,6 +150,7 @@ export const HeadingWithAnchor: FunctionComponent<
   React.PropsWithChildren<HeadingWithAnchorProps>
 > = ({ tag, anchor: propsAnchor, hideAnchor, children, tagProps }) => {
   const isInSidebar = React.useContext(IsSidebarContext)
+  const isInDialog = React.useContext(IsDialogContext)
```

Comment:
I agree that there is room for improvement over what we do in the PR and we should update it in a follow-up!
I have thought about merging the contexts but one issue I saw was that `isInSidebar` is used in other places, too, in order to influence styling, e.g. in [StyledStreamlitMarkdown here](https://github.com/streamlit/streamlit/blob/develop/frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts#L46) in order to what looks like making links look good in the sidebar etc. Though in Dialogs we want the "normal" look and feel of links.

Another approach might be to try to lift the contexts up and rather push the modified styling down (which might be tricky because sidebar/dialog do not know their children) or provide properties / contexts that are more detached from the Streamlit concepts such as sidebar or dialog (e.g. `NoAnchorsContext` / `noAnchors=true`, `PlainLinksContext` / `plainLinks=true`, ...) and then those contexts / props could be set more fine-granular by the sidebar or dialog. The advantage would be that we detangle the logic and make the components more re-usable, because `isInSidebar`/`isInDialog` is very Streamlit webapp related, whereas `noAnchors` is meaningful independent of the Streamlit webapp (thinking about the overall separation effort). 
This might need a little bit more cogitating about how to do right, but those were my thoughts here so far 🙂
What do you think?

Reply:
Yea. That definitely makes sense. It shows what kinds of "settings" we can apply to any containers and give new concepts to manage.

Let's file that as tech debt, and consider it for the next container.